src/Expr.js

```diff
@@ -169,35 +236,109 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    var argumentsExpr = null
+
+    if (isExpr(expr['arguments'])) {
+      argumentsExpr = exprToString(expr['arguments'])
+    } else {
+      if (Array.isArray(expr['arguments'])) {
+        argumentsExpr = '[' + printArray(expr['arguments'], exprToString) + ']'
+      } else if (typeof expr['arguments'] === 'object') {
+        argumentsExpr = printObject(expr['arguments'])
+      } else {
+        argumentsExpr = exprToString(expr['arguments'])
+      }
+    }
+
+    var callExpr = ''
+    if (isExpr(expr['call'])) {
```

Comment:
no need to check if `expr['call']` is an expression, the recursion will already do this

Reply:
Ah, true! Thanks. Will update.